src/main/java/org/repodriller/scm/GitRepository.java

```diff
@@ -212,77 +206,87 @@ private GregorianCalendar convertToDate(RevCommit revCommit) {
-
-				List<DiffEntry> diffsForTheCommit = diffsForTheCommit(repo, jgitCommit);
-				if (diffsForTheCommit.size() > this.getMaxNumberFilesInACommit()) {
-					log.error("commit " + id + " has more than files than the limit");
-					throw new RuntimeException("commit " + id + " too big, sorry");
-				}
+			if (!itr.hasNext())
+				return null;
+			RevCommit jgitCommit = itr.next();
+
+			/* Extract metadata. */
+			Developer author = new Developer(jgitCommit.getAuthorIdent().getName(), jgitCommit.getAuthorIdent().getEmailAddress());
+			Developer committer = new Developer(jgitCommit.getCommitterIdent().getName(), jgitCommit.getCommitterIdent().getEmailAddress());
+			TimeZone authorTimeZone = jgitCommit.getAuthorIdent().getTimeZone();
+			TimeZone committerTimeZone = jgitCommit.getCommitterIdent().getTimeZone();
 
-				for (DiffEntry diff : diffsForTheCommit) {
+			String msg = jgitCommit.getFullMessage().trim();
+			String hash = jgitCommit.getName().toString();
+			String parent = (jgitCommit.getParentCount() > 0) ? jgitCommit.getParent(0).getName().toString() : "";
 
-					ModificationType change = Enum.valueOf(ModificationType.class, diff.getChangeType().toString());
+			GregorianCalendar authorDate = new GregorianCalendar();
+			authorDate.setTime(jgitCommit.getAuthorIdent().getWhen());
+			authorDate.setTimeZone(jgitCommit.getAuthorIdent().getTimeZone());
 
-					String oldPath = diff.getOldPath();
-					String newPath = diff.getNewPath();
+			GregorianCalendar committerDate = new GregorianCalendar();
+			committerDate.setTime(jgitCommit.getCommitterIdent().getWhen());
+			committerDate.setTimeZone(jgitCommit.getCommitterIdent().getTimeZone());
 
-					String diffText = "";
-					String sc = "";
-					if (diff.getChangeType() != ChangeType.DELETE) {
-						diffText = getDiffText(repo, diff);
-						sc = getSourceCode(repo, diff);
-					}
+			boolean isMerge = (jgitCommit.getParentCount() > 1);
 
-					if (diffText.length() > maxSizeOfDiff) {
-						log.error("diff for " + newPath + " too big");
-						diffText = "-- TOO BIG --";
-					}
-					
-					theCommit.addModification(oldPath, newPath, change, diffText, sc);
+			Set<String> branches = getBranches(git, hash);
+			boolean isCommitInMainBranch = branches.contains(this.mainBranchName);
 
+			/* Create one of our Commit's based on the jgitCommit metadata. */
+			Commit commit = new Commit(hash, author, committer, authorDate, authorTimeZone, committerDate, committerTimeZone, msg, parent, isMerge, branches, isCommitInMainBranch);
+
+			/* Convert each of the associated DiffEntry's to a Modification. */
+			List<DiffEntry> diffsForTheCommit = diffsForTheCommit(repo, jgitCommit);
+			if (diffsForTheCommit.size() > maxNumberFilesInACommit) {
+				/* TODO I don't understand why this is exceptional. We should mark the Commit as truncated and handle the first X files. */
```

Comment:
Some repos have huge commits that can give you OOM. That's why we enable users to filter out commits that are too long. 

In addition, I believe ignoring is better than analyzing the X first files, as you'd analyze an incomplete commit, which may make your data a bit dirty.

Reply:
You're right, I guess a partial commit is worse than no commit.

I'm not entirely satisfied, however, with the "use environment variables to control behavior" approach. It works as long as there are very few knobs, but perhaps adding a configuration file might be a good Issue?

Reply:
I am avoiding config files because the 'beauty' of repodriller is to have a DSL that allows you to configure everything. Maybe we can add a method in the `RepositoryMining` DSL!

Reply:
At the moment this aspect of the configuration is done with environment variables. I was suggesting we replace the env vars with a config file.

Are you suggesting that they should instead be replaced with APIs?